### ISSUE TITLE
Interested in becoming a maintainer

### ISSUE THREAD

**Issue by darshit-desai** [dronekit/dronekit-python#1196]
Hi developer team,

I am interested in becoming a contributor for this repository and here are my experiences with dronekit, Ardupilot and PX4 which helps me better understand the improvements which might come up in the future,

I have used dronekit python earlier, to develop an airdrop selection script which used data streamed (airspeed, ground speed, orientation) through mavlink to actuate the servo of the bottle drop.
I have used Mavros and PX4 to implement an offboard obstacle avoidance script, which takes data from d435i stereo camera and orients the drone away from the obstacle based on the pose of the obstacle.
I have also worked with the core hardware and know about the communication protocols of the RC hardware.

Thank you for considering me,
Best,
Darshit

**Comment by hamishwillee** [comment]
Thanks @darshit-desai 

@morzack is volunteered recently too, and is co-ordinating reviews of PRs and issues. Please collaborate. I would like you to work together on a few reviews and also co-ordinate with @peterbarker before giving you merge rights.

**Comment by darshit-desai** [comment]
Thank you @hamishwillee, I sent an email to @morzack. Is there a place where I can get started, I come from the user side where I only used Dronekit as an API to write apps and not really know what actually goes on under the hood.

**Comment by morzack** [comment]
Hi, I just replied!

Just for transparency here, I mentioned how I've been going through the open issues and categorizing them into several buckets (stale, urgently needs fixing, etc...).

@hamishwillee in your opinion is reviewing the currently open PRs the most relevant first step (as opposed to any dev work on issues)?

**Comment by hamishwillee** [comment]
@morzack My opinion is that getting DroneKit rock solid in Python 3 should be the "project priority".

But mostly I want to keep the project alive and lively, so what I'm after is professionalism, process, common sense. As long as we see that, it should be the priorities of the people actually doing the work that is most important - i.e. you.

**Comment by TongBingda** [comment]
Hi, developer team of dronekit project,

I'm a developer from China, I've added some new features in the last few days and I'm interested in becoming a contributor to this repository. I am currently a PhD student. I have been working with dronekit and ardupilot for UAS development since 2020, which helps me to better participate in future updates of the project.I am currently using drone python to develop a lightweight drone cluster ground station script that obtains drone information and controls drone state by using the pysimplegui library and the dronekit library. I have also used a Raspberry Pi and NVIDIA jetson NX as companion computers to control quadcopters and fixed wing drones. I have also used remote controllers like the radiomaster tx16s that have openTX or edgeTX open source operating systems installed.

I have emailed @morzack . Thank you for considering me!

**Comment by morzack** [comment]
Hi @TongBingda , feel free to make a pull request with the changes and then it can be tested + reviewed

**Comment by hamishwillee** [comment]
So any progress on this?. Would be good to see some PRs for concrete Python 3 support.

**Comment by hamishwillee** [comment]
We've had a number of people express interest in becoming maintainers -  @morzack @a87chaud @alymkarim @Ian-VA @Russi1555 @Tuxliri @abdulbasit189  (related discussions, which I will close, are #1240, #1237, #1228, #1225, 1242).

To be clear there are no maintainers currently - I used to work on the documentation but I have moved on. This is still used by the ArduPilot community. 

The next main steps required are to get this working **solidly** with Python 3, make a release, and update the documentation. 

In the meantime I would suggest that if you are interested in helping you start off my looking at existing issues in pairs or more and see if you can agree on fixes. Once we have a few people with some proven expertise and ongoing commitment we can move forward on a solid base.

**Comment by alymkarim** [comment]
> We've had a number of people express interest in becoming maintainers - [@morzack](https://github.com/morzack) [@a87chaud](https://github.com/a87chaud) [@alymkarim](https://github.com/alymkarim) [@Ian-VA](https://github.com/Ian-VA) [@Russi1555](https://github.com/Russi1555) [@Tuxliri](https://github.com/Tuxliri) (related discussions, which I will close, are [#1240](https://github.com/dronekit/dronekit-python/issues/1240), [#1237](https://github.com/dronekit/dronekit-python/issues/1237), [#1228](https://github.com/dronekit/dronekit-python/issues/1228), [#1225](https://github.com/dronekit/dronekit-python/issues/1225)).
> 
> To be clear there are no maintainers currently - I used to work on the documentation but I have moved on. This is still used by the ArduPilot community.
> 
> The next main steps required are to get this working **solidly** with Python 3, make a release, and update the documentation.
> 
> In the meantime I would suggest that if you are interested in helping you start off my looking at existing issues in pairs or more and see if you can agree on fixes. Once we have a few people with some proven expertise and ongoing commitment we can move forward on a solid base.

yep that sounds good to me, should we start to work on it? @morzack  @a87chaud @Ian-VA @Russi1555 @Tuxliri

**Comment by a87chaud** [comment]
Hi @alymkarim  I would be happy to work on any issue.
@hamishwillee is there any critical issue we should start off with? it seems there is security issue #1239 which we could start off with.

**Comment by hamishwillee** [comment]
I think you should work together to find issues you think are important. Personally I think that security issue is low priority. My reasoning is that very few people use IP endpoints in dronekit but many use everything else.

I consider any bugs that prevent this running on Python 3.7 or the latest python (whatever that is) to be the important ones. Once we have fixed all those we'd be in a reasonable position to cut a new release.

**Comment by a87chaud** [comment]
There doesnt seem to be support for sitl for python3 would that be a good starting point?

**Comment by hamishwillee** [comment]
Yes.
There is "initial support" for Python 3 in the main branch (i.e. not a release). Getting that robust would be a great first step.

And as part of that, having dronekit sitl that works on Python3 would be awesome. I don't know the state of that, if anything.

**Comment by a87chaud** [comment]
Cool! I will read the documentation and get more familiar with it. Any tips/resources on how to get started?

**Comment by hamishwillee** [comment]
One thing you might do is work through all the examples with Python 3. That will show you how things are expected to work, and immediately highlight the problems with both the examples and dronekit.

So you could fix both as you go - docs examples and DK python.

**Comment by Optimized-Brain** [comment]
Hi @hamishwillee , I am also interested in contributing to this project. Please consider.

**Comment by hamishwillee** [comment]
@Optimized-Brain Thanks! See my comment https://github.com/dronekit/dronekit-python/issues/1196#issuecomment-3134720530 - nothing has changed. We'd welcome contributions but you guys are going to need to work together on this.

**Comment by Optimized-Brain** [comment]
Okay thank you I will keep note of it 👍
